modules/standard/Map.chpl

```diff
@@ -401,16 +405,18 @@ module Map {
     */
     iter values() ref
     where !isNonNilableClass(valType) {
-      for val in vals {
-        yield val;
+      for slot in table.allSlots() {
+        if table.isSlotFull(slot) then
+          yield table.table[slot].val;
       }
     }
 
     pragma "no doc"
     iter values() const where isNonNilableClass(valType) {
```

Comment:
I think that with `chpl__hashtable` as the implementation, it should be possible to remove this overload.